toucan_connectors/snowflake_common.py

```diff
@@ -171,28 +171,36 @@ def _execute_parallelized_queries(
 
     def fetch_data(
         self,
-        connection,
+        connection: SnowflakeConnection,
         data_source: SfDataSource,
         offset: Optional[int] = None,
         limit: Optional[int] = None,
         get_row_count: bool = False,
     ) -> pd.DataFrame:
-        self._execute_query(connection, f'USE DATABASE {data_source.database}')
-        self._execute_query(connection, f'USE WAREHOUSE {data_source.warehouse}')
+        prepare_query: List = []
+        if data_source.database != connection.database:
+            prepare_query.append(f'USE DATABASE {data_source.database}')
+        if data_source.warehouse != connection.warehouse:
+            prepare_query.append(f'USE WAREHOUSE {data_source.warehouse}')
+        if len(prepare_query) > 0:
+            self._execute_query(connection, ';'.join(prepare_query) + ';')
```

Comment:
Nice, just a proposition, of a short way, with ternaries... 
Without using list... :
```python
self._execute_query(connection, 
    (f'USE DATABASE {data_source.database};' if data_source.database != connection.database else '') + \
    (f'USE WAREHOUSE {data_source.warehouse};' if data_source.warehouse != connection.warehouse else '')
);
```

Reply:
What is the advantage of this solution over using a list?

Reply:
Actually, there is not a huge advantage with this approach... but we have : 
- less lines of code.
- no variables (it's not a big deal, but the creation of a list, the append on it and the join on it can be small in term of resources for one call in dev.. but for thousands of calls in prod... )
- if we're going to add more prepared queries in the future, we will prevent manipulating variables(append/join) and directly concatenate the final query depending on conditions... with less lines.


i assume it's not a big deal after all 😅 , but in terms of  `'optimizations'` it can be nice.... i guess.... ?